lib/Sema/SemaAPINotes.cpp

```diff
@@ -180,7 +179,7 @@ static void ProcessAPINotes(Sema &S, Decl *D,
         return UnavailableAttr::CreateImplicit(S.Context,
                                                CopyString(S.Context,
                                                           info.UnavailableMsg));
-    });
+    }, [](Decl *decl) { return decl->specific_attr_begin<UnavailableAttr>(); });
```

Comment:
Really not happy with this duplication. If MSVC really can't handle the default argument, maybe you can make an overload instead?